packages/react-components/react-tree/src/utils/createUnfilteredFlatTree.ts

```diff
@@ -0,0 +1,81 @@
+import { FlatTreeItem, FlatTreeItemProps } from '../hooks/useFlatTree';
+
+/**
+ * @internal
+ */
+export type UnfilteredFlatTree = {
+  items: FlatTreeItem[];
+  itemsPerId: Map<string, FlatTreeItem>;
+};
+
+/**
+ * creates a list of flat tree items
+ * and provides a map to access each item by id
+ */
+export function createUnfilteredFlatTree(flatTreeItemProps: FlatTreeItemProps[]): UnfilteredFlatTree {
+  const root = createRootFlatTreeItem();
+  const itemsPerId = new Map<string, FlatTreeItem>([[root.id, root]]);
+  const items: FlatTreeItem[] = [];
+
+  for (let index = 0; index < flatTreeItemProps.length; index++) {
+    const { parentId = flatTreeRootParentId, ...treeItemProps } = flatTreeItemProps[index];
+
+    const nextItemProps = flatTreeItemProps[index + 1] as FlatTreeItemProps | undefined;
+    const currentParent = itemsPerId.get(parentId);
+    if (!currentParent) {
+      if (process.env.NODE_ENV === 'development') {
```

Comment:
I know this is merged already but this is probably better written as

`if (process.env.NODE_ENV !== 'production') {`

so this warning will appear in testing scenarios. This is also aligns with how we do this [elsewhere](https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-utilities/src/hooks/useControllableState.ts#L81).